### DISCUSSION TITLE
Making one CSV from multiple tables MIMIC - III

### DISCUSSION THREAD

**Discussion by Ioanni-Balassis** [MIT-LCP/mimic-code#1386] [MIMIC-III] [answered]
I am looking for a way to create a CSV table with the following columns 
HADM_ID
ADMITTIME 
DISCHTIME
TEXT

The text being any type of text (preferably the patients history) related to that admission. 
This is to be used in order to try predict length of stay using NLP

**Comment by marymlucas** [comment, marked as answer]
Hi @Ioanni-Balassis,
If working from the csv files my approach would be to use the Python Pandas library or alternatively you can use R dplyr library. See some sample Python code below that should work, here I am getting the text field from the noteevents table. 


```
import pandas as pd
admissions = pd.read_csv('ADMISSIONS.csv.gz')
noteevents = pd.read_csv('NOTEEVENTS.csv.gz')
df = pd.merge(admissions[['HADM_ID', 'ADMITTIME', 'DISCHTIME']], noteevents[['HADM_ID','TEXT']], on='HADM_ID', how='left')
df.to_csv('my_results.csv', index=False)
```


If working from BigQuery then you can write a query to join the admissions table with the noteevents table on HADM_ID using the same logic as above, and then export the results to CSV.

Not sure if this is what you're looking for but hope it's helpful
Mary
